### PR TITLE
Grammar fixes in `control-structures.rst`

### PR DESCRIPTION
**Pull request by krakxn**
## Category:

Documentation

## Description:

Grammatical corrections.

#### Doggo

images

### CODE REVIEW

File: docs/control-structures.rst [hunk 1]
```diff
@@ -101,18 +101,18 @@ The ``@nonreentrant(<key>)`` decorator places a lock on a function, and all func
         # this function is protected from re-entrancy
         ...
 
-You can put the ``@nonreentrant(<key>)`` decorator on a ``__default__`` function but we recommend against it because in most circumstances it will not work in a meaningful way.
+You can put the ``@nonreentrant(<key>)`` decorator on a ``__default__`` function, but we recommend against it because in most circumstances it will not work as intended.
```
Comment (charles-cooper) [inline docs/control-structures.rst:104]: not sure why this change was made?

File: docs/control-structures.rst [hunk 2]
```diff
@@ -122,7 +122,7 @@ A contract can also have a default function, which is executed on a call to the
 
 This function is always named ``__default__``. It must be annotated with ``@external``. It cannot expect any input arguments.
 
-If the function is annotated as ``@payable``, this function is executed whenever the contract is sent Ether (without data). This is why the default function cannot accept arguments - it is a design decision of Ethereum to make no differentiation between sending ether to a contract or a user address.
+If the function is annotated as ``@payable``, this function is executed whenever the contract is sent Ether (without data). This is why the default function cannot accept arguments — it is a design decision of Ethereum to make no differentiation between sending Ether to a contract or a user address.
```
Comment (charles-cooper) [inline docs/control-structures.rst:125]: does the different in hyphen show up in the final output?

File: docs/control-structures.rst (hunk 1 above)
Comment (krakxn) [reply docs/control-structures.rst:104]: Actually, even the correction is wrong. It should be the following instead:

```diff
+You can put the ``@nonreentrant(<key>)`` decorator on a ``__default__`` function; but we recommend against it, because in most circumstances it will not work as intended.
```
Rationality: 

- `You can put the ``@nonreentrant(<key>)`` decorator on a ``__default__`` function;` is an independent clause, and leads to a comma splice error. To avoid this, we add a `;`
- `because in most circumstances it will not work as intended.` is a relative clause which adds to the first part (or `but we recommend against it,` so we add a comma here).
- As for the change to `as intended` -- it is so for concision.

File: docs/control-structures.rst (hunk 2 above)
Comment (krakxn) [reply docs/control-structures.rst:125]: It does so when I preview it in VSCode.

File: docs/control-structures.rst [hunk 3]
```diff
@@ -101,18 +101,18 @@ The ``@nonreentrant(<key>)`` decorator places a lock on a function, and all func
...
 
-Nonreentrancy locks work by setting a specially allocated storage slot to a ``<locked>`` value on function entrance, and setting it to an ``<unlocked>`` value on function exit. On function entrance, if the storage slot is detected to be the ``<locked>`` value, execution reverts.
+Non-reentrancy locks work by setting a specially allocated storage slot to a ``<locked>`` value on function entrance, and setting it to an ``<unlocked>`` value on function exit. On function entrance, if the storage slot is detected to be the ``<locked>`` value, execution reverts.
 
-You cannot put the ``@nonreentrant`` decorator on a ``pure`` function. You can put it on a ``view`` function, but it only checks that the function is not in a callback (the storage slot is not in the ``<locked>`` state), as ``view`` functions can only read the state, not change it.
+You cannot put the ``@nonreentrant`` decorator on a ``pure`` function. You can put it on a ``view`` function, but it only checks that the function is not in a callback (the storage slot is not in the ``<locked>`` state), as ``view`` functions can only read the state and cannot change it.
```
Comment (charles-cooper) [inline docs/control-structures.rst:108]: I don't think this change is necessary

File: docs/control-structures.rst [hunk 4]
```diff
@@ -158,7 +158,7 @@ Lastly, although the default function receives no arguments, it can still access
 The ``__init__`` Function
 -----------------------
 
-``__init__`` is a special initialization function that may only be called at the time of deploying a contract. It can be used to set initial values for storage variables. A common use case is to set an ``owner`` variable with the creator the contract:
+``__init__`` is a special initialization function that may only be called at the time of deploying a contract. It can be used to set initial values for storage variables. A common use-case is to set an ``owner`` variable with the creator of the contract:
```
Comment (charles-cooper) [inline docs/control-structures.rst:161]: I think unhyphenation here is a better style

**Review by charles-cooper — CHANGES_REQUESTED** [review]
